src/test.js

```diff
@@ -732,18 +732,13 @@ function internalStart( test ) {
 	}
 }
 
-function numberOfTests( module ) {
-	let count = module.tests.length,
-		modules = [ ...module.childModules ];
+function numberOfTests( module, offset = 0 ) {
 
 	// Do a breadth-first traversal of the child modules
-	while ( modules.length ) {
-		let nextModule =  modules.shift();
-		count += nextModule.tests.length;
-		modules.push( ...nextModule.childModules );
-	}
-
-	return count;
+	return module.childModules.reduce(
+		( p, childModule ) => numberOfTests( childModule, p ),
+		module.tests.length + offset
```

Comment:
This is pretty confusing. I'd put the arguments on the previous line, as usual with other callback methods, and keep the body on its own line, adding curely braces as necessary.